### PR TITLE
Add benchmark latency test

### PR DESCRIPTION
**Pull request by maayanf24**
Fixes issue #677

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

File: pkg/subctl/benchmark/latency.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func StartLatencyTests() {
+	gomega.RegisterFailHandler(func(message string, callerSkip ...int) { panic(message) })
+	f := initFramework("latency")
+	framework.By("Performing latency tests from Gateway pod to Gateway pod")
+	runLatencyTest(f, framework.GatewayNode)
+	framework.By("Performing latency tests from Non-Gateway pod to Non-Gateway pod")
+	runLatencyTest(f, framework.NonGatewayNode)
+	cleanupFramework(f)
```
Comment (tpantelis) [inline pkg/subctl/benchmark/latency.go:18]: Can you add blank lines between logical sections to improve readability? I'd suggest one before L14, L16, and L18.

File: pkg/subctl/cmd/benchmark.go [hunk 2]
```diff
@@ -22,10 +22,20 @@ var (
 		},
 		Run: testThroughput,
 	}
+	benchmarkLatenchCmd = &cobra.Command{
+		Use:   "latency <kubeconfig1> <kubeconfig2>",
+		Short: "Benchmark latency between two clusters",
+		Long:  "This command run latency tests between two clusters",
```
Comment (tpantelis) [inline pkg/subctl/cmd/benchmark.go:28]:
```suggestion
		Long:  "This command runs latency benchmark tests between two clusters",
```

File: pkg/subctl/benchmark/latency.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"fmt"
+
+	"github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartLatencyTests() {
+	gomega.RegisterFailHandler(func(message string, callerSkip ...int) { panic(message) })
```
Comment (tpantelis) [inline pkg/subctl/benchmark/latency.go:12]: Shouldn't we run `cleanupFramework(f) ` on assertion failure?

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/subctl/benchmark/latency.go (hunk 3 above)
Comment (mangelajo) [reply pkg/subctl/benchmark/latency.go:12]: Oh right, Let's move this line after L13, and call cleanupFramework before the panic.

**Review by mangelajo — CHANGES_REQUESTED** [review]
See tom comments

File: pkg/subctl/benchmark/latency.go (hunk 3 above)
Comment (maayanf24) [reply pkg/subctl/benchmark/latency.go:12]: the RegisterFailHandler() need to be called before framework init

Comment (maayanf24) [reply pkg/subctl/benchmark/latency.go:12]: the RegisterFailHandler() need to be called before framework init

**Review by tpantelis — APPROVED** [review]

File: pkg/subctl/benchmark/latency.go (hunk 3 above)
Comment (mangelajo) [reply pkg/subctl/benchmark/latency.go]: You could use a variable for the framework

`var f framework.Framework`  before the gomega.RegisterFailHandler


Then 

`f = initFramework("latency")`

**Review by mangelajo — CHANGES_REQUESTED** [review]

**Review by skitt — APPROVED** [review]

**Comment by mangelajo** [comment]
Reading the updated version sorry, I didn't refresh

File: pkg/subctl/benchmark/latency.go [hunk 4]
```diff
@@ -9,12 +9,19 @@ import (
 )
 
 func StartLatencyTests() {
-	gomega.RegisterFailHandler(func(message string, callerSkip ...int) { panic(message) })
+	gomega.RegisterFailHandler(func(message string, callerSkip ...int) {
+		framework.RunCleanupActions()
+		panic(message)
+	})
+
```
Comment (mangelajo) [inline pkg/subctl/benchmark/latency.go:16]:
```suggestion
	var f *framework.Framework
	
	gomega.RegisterFailHandler(func(message string, callerSkip ...int) {
		if f != nil {
			cleanupFramework(f)
		} else {
			framework.RunCleanupActions()
		}
		panic(message)
	})

```

File: pkg/subctl/benchmark/latency.go [hunk 5]
```diff
@@ -9,12 +9,19 @@ import (
...
 	f := initFramework("latency")
```
Comment (mangelajo) [inline pkg/subctl/benchmark/latency.go:17]:
```suggestion
	f = initFramework("latency")
```

**Review by mangelajo — CHANGES_REQUESTED** [review]

File: pkg/subctl/benchmark/latency.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartLatencyTests() {
+	var f *framework.Framework
+
+	gomega.RegisterFailHandler(func(message string, callerSkip ...int) {
```
Comment (mangelajo) [inline pkg/subctl/benchmark/latency.go:14]: I guess that we will need the same on the benchmark patch. Can you open an PR for it?

**Review by mangelajo — APPROVED** [review]

**Review by skitt — APPROVED** [review]

File: pkg/subctl/benchmark/latency.go (hunk 6 above)
Comment (maayanf24) [reply pkg/subctl/benchmark/latency.go:14]: Sure, I'll open a PR

**Pull request merged**